### ISSUE TITLE
google_compute_instance should allow to specify instance state

### ISSUE THREAD

**Issue by roman8422** [hashicorp/terraform-provider-google#1719]
Hi,

There are several long lasting issues open with the same functionality request for aws_instance.
It would be very useful to have such option for google_compute_instance as well.

### Terraform Version
Terraform v0.11.7

### Affected Resource(s)
- google_compute_instance

### Terraform Configuration Files
```hcl
resource "google_compute_instance" "default" {
  name         = "test"
  machine_type = "f1-micro"
  zone         = "us-central1-a"
# state        = "running"
# state        = "stopped"

  boot_disk {
    initialize_params {
      image = "debian-cloud/debian-8"
    }
  }

  network_interface {
    network = "default"
  }
}
```

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- https://github.com/terraform-providers/terraform-provider-aws/issues/22
- https://github.com/terraform-providers/terraform-provider-aws/issues/190

**Comment by paddycarver** [comment]
Hi, thanks for the feature request! As you noted, this is a common feature request for other providers, and I think the current state of things is best summed up by [Mitchell's response](https://github.com/hashicorp/terraform/issues/1579#issuecomment-95117648) to the feature request for AWS: managing the state of an instance is more complicated than managing the presence of an instance (how do I go from "terminated" to "stopped"? What do we even do with "suspended"?)  I think upstream Terraform was thinking on some of this a bit ago, but I don't know that a decision was ever reached.

**Comment by hounded** [comment]
This is so needed! I tried deploying the new gitlab chart to a cluster running  2 `n1-standard-1` ($50 a month) and the resource hungry beast wouldn't run, it did run on 2 `n1-standard-4` but I can't foot a $300 a month bill for my devops when I'm a solo developer, would be so handy to be able to shutdown node pools and then bring them up when needed.

**Comment by pondohva** [comment]
@paddycarver I'm not really sure why do we have "STOPPED" status in compute client. https://cloud.google.com/compute/docs/instances/instance-life-cycle describes TERMINATED state as state for stopped instance. I've created PR using life cycle info from this doc.

**Comment by upodroid** [comment]
@rileykarson We can close this. It was fixed in #4797
